packages/twenty-server/src/engine/core-modules/client-config/services/client-config.service.ts

```diff
@@ -0,0 +1,144 @@
+        password: this.twentyConfigService.get('AUTH_PASSWORD_ENABLED'),
+        microsoft: this.twentyConfigService.get('AUTH_MICROSOFT_ENABLED'),
+        sso: [],
+      },
+      signInPrefilled: this.twentyConfigService.get('SIGN_IN_PREFILLED'),
+      isMultiWorkspaceEnabled: this.twentyConfigService.get(
+        'IS_MULTIWORKSPACE_ENABLED',
+      ),
+      isEmailVerificationRequired: this.twentyConfigService.get(
+        'IS_EMAIL_VERIFICATION_REQUIRED',
+      ),
+      defaultSubdomain: this.twentyConfigService.get('DEFAULT_SUBDOMAIN'),
+      frontDomain: this.domainManagerService.getFrontUrl().hostname,
+      debugMode:
+        this.twentyConfigService.get('NODE_ENV') ===
+        NodeEnvironment.development,
+      support: {
+        supportDriver: supportDriver
+          ? this.transformEnum(supportDriver, SupportDriver)
+          : SupportDriver.None,
+        supportFrontChatId: this.twentyConfigService.get(
+          'SUPPORT_FRONT_CHAT_ID',
+        ),
+      },
+      sentry: {
+        environment: this.twentyConfigService.get('SENTRY_ENVIRONMENT'),
+        release: this.twentyConfigService.get('APP_VERSION'),
+        dsn: this.twentyConfigService.get('SENTRY_FRONT_DSN'),
+      },
+      captcha: {
+        provider: captchaProvider
+          ? this.transformEnum(captchaProvider, CaptchaDriverType)
+          : undefined,
+        siteKey: this.twentyConfigService.get('CAPTCHA_SITE_KEY'),
+      },
+      chromeExtensionId: this.twentyConfigService.get('CHROME_EXTENSION_ID'),
+      api: {
+        mutationMaximumAffectedRecords: this.twentyConfigService.get(
+          'MUTATION_MAXIMUM_AFFECTED_RECORDS',
+        ),
+      },
+      isAttachmentPreviewEnabled: this.twentyConfigService.get(
+        'IS_ATTACHMENT_PREVIEW_ENABLED',
+      ),
+      analyticsEnabled: this.twentyConfigService.get('ANALYTICS_ENABLED'),
+      canManageFeatureFlags:
+        this.twentyConfigService.get('NODE_ENV') ===
+          NodeEnvironment.development ||
+        this.twentyConfigService.get('IS_BILLING_ENABLED'),
+      publicFeatureFlags: PUBLIC_FEATURE_FLAGS,
+      isMicrosoftMessagingEnabled: this.twentyConfigService.get(
+        'MESSAGING_PROVIDER_MICROSOFT_ENABLED',
+      ),
+      isMicrosoftCalendarEnabled: this.twentyConfigService.get(
+        'CALENDAR_PROVIDER_MICROSOFT_ENABLED',
+      ),
+      isGoogleMessagingEnabled: this.twentyConfigService.get(
+        'MESSAGING_PROVIDER_GMAIL_ENABLED',
+      ),
+      isGoogleCalendarEnabled: this.twentyConfigService.get(
+        'CALENDAR_PROVIDER_GOOGLE_ENABLED',
+      ),
+      isConfigVariablesInDbEnabled: this.twentyConfigService.get(
+        'IS_CONFIG_VARIABLES_IN_DB_ENABLED',
+      ),
+    };
+
+    return clientConfig;
+  }
+
+  // GraphQL enum values are in PascalCase, but the config values are in kebab-case
+  // This function transforms the config values, the same way GraphQL does
+  private transformEnum<T extends Record<string, string>>(
```

Comment:
A few thought about this one:
1) is this a breaking change we should be aware of?
2) Also, why do we talk about GraphQL here as we are in a service, does not feel to be the right level
3) I'm not sure about why we transformEnum but if it's important it seems to be a broader issue than client-config and should likely be implemented globally